pkg/metrics/metrics.go

```diff
@@ -95,16 +104,37 @@ func (metrics *metrics) CountResult(result api.Result) {
 	metrics.Completed.With(prometheus.Labels{"result": string(result)}).Inc()
 }
 
-// ObserveDurationByState logs duration of the state
-func (metrics *metrics) ObserveDurationByState(state *api.StateItem) error {
+// ObserveTotalDurationByState logs duration of the given state
+func (metrics *metrics) ObserveTotalDurationByState(state *api.StateItem) error {
 	if state.StartedAt.IsZero() {
 		return fmt.Errorf("cannot observe StateItem if StartedAt is not set")
 	}
 	duration := state.FinishedAt.Sub(state.StartedAt.Time)
 	if duration < 0 {
 		return fmt.Errorf("cannot observe StateItem if FinishedAt is before StartedAt")
 	}
-	metrics.Duration.With(prometheus.Labels{"state": string(state.State)}).Observe(duration.Seconds())
+	metrics.TotalDuration.With(prometheus.Labels{"state": string(state.State)}).Observe(duration.Seconds())
+	return nil
+}
+
+// ObserveCurrentDurationByState logs duration of the current and unfinished pipeline state
+func (metrics *metrics) ObserveCurrentDurationByState(run *api.PipelineRun) error {
+	var duration time.Duration
+	if run.Status.State == api.StateUndefined {
+		duration = time.Now().Sub(run.CreationTimestamp.Time)
+		if duration < 0 {
+			return fmt.Errorf("cannot observe pipelinerun if CreationTimestamp is before StartedAt of the state %v", run.Status.State)
+		}
```

Comment:
`StateNew` incorporates `StateUndefined` (see [here](https://github.com/SAP/stewardci-core/blob/c3aebbcbe660fd3722f2f00dcf3b253c43563d46/pkg/k8s/pipelineRun.go#L138-L158)). We don't need to differentiate metrics for `StateUndefined` and `StateNew`, because the controller doesn't start processing a pipeline run object before changing to `StatePreparing`. Therefore, it would be appropriate to label with `StateNew` instead of the empty string.

Reply:
I had this logic for the case a pipelinerun will be never processed by the controller somehow.

Reply:
It's required to count also pipeline runs where no state is set. In such case, use label `state="new"` instead of `state=""`.

Reply:
done